registry/handlers/app.go

```diff
@@ -194,6 +195,28 @@ func NewApp(ctx context.Context, config *configuration.Configuration) *App {
 		}
 	}
 
+	// configure foreign layers
+	if flConfig, ok := config.Storage["foreignlayers"]; ok {
+		if hosts, ok := flConfig["hostwhitelist"]; ok {
+			switch hosts := hosts.(type) {
+			case []interface{}:
```

Comment:
With the current code I don't think this would work, but you could probably work around it with something like `REGISTRY_STORAGE_FOREIGNLAYER="hostwhitelist: [a.com]"`.

We don't have a good general solution for expressing array values in environment variables.